instrumentation/src/androidTest/java/com/bumptech/glide/NonBitmapDrawableResourcesTest.java

```diff
@@ -31,19 +31,29 @@
 import org.junit.function.ThrowingRunnable;
 import org.junit.rules.TestName;
 import org.junit.runner.RunWith;
-import org.mockito.MockitoAnnotations;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
 
-@RunWith(AndroidJUnit4.class)
+@RunWith(Parameterized.class)
 public class NonBitmapDrawableResourcesTest {
   @Rule public final TestName testName = new TestName();
   @Rule public final TearDownGlide tearDownGlide = new TearDownGlide();
 
+  @Parameter public boolean useDirectResourceLoader;
+
+  @Parameters(name = "useDirectResourceLoader = {0}")
+  public static ImmutableList<Boolean> parameters() {
+    return ImmutableList.of(true, false);
+  }
```

Comment:
**Suggestion**: Ensure that the parameterized test setup with useDirectResourceLoader is thoroughly validated across different test scenarios to confirm that both true and false states behave as expected without unintended side effects. <br> **Code Suggestion**: 
 ```
 + // Additional validation or setup might be required for comprehensive testing
 ```